repo/blob/s3/s3_assume_role.go

```diff
@@ -0,0 +1,83 @@
+package s3
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awscreds "github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/pkg/errors"
+)
+
+// minioProvider is a shim that implements the Minio `Provider` interface
+// for an AWS credential.
+type minioProvider struct {
+	creds *awscreds.Credentials
+}
+
+func assumeRoleCredentials(
+	roleARN string,
+	roleSessionName string,
+	duration string,
+	tags map[string]string,
+) (credentials.Provider, error) {
+	var (
+		roleDuration time.Duration
+		err          error
+	)
+
+	if duration != "" {
+		roleDuration, err = time.ParseDuration(duration)
+		if err != nil {
+			return &minioProvider{}, errors.Wrap(err, "NewSession")
+		}
+	}
+
+	sess, err := session.NewSession()
+	if err != nil {
+		return &minioProvider{}, errors.Wrap(err, "ParseDuration")
+	}
+
```

Comment:
The error messages are mixed up.

```suggestion
		roleDuration, err = time.ParseDuration(duration)
		if err != nil {
			return &minioProvider{}, errors.Wrap(err, "ParseDuration")
		}
	}

	sess, err := session.NewSession()
	if err != nil {
		return &minioProvider{}, errors.Wrap(err, "NewSession")
	}

```